lib/dependencies/AMDRequireDependenciesBlockParserPlugin.js

```diff
@@ -46,8 +46,116 @@ class AMDRequireDependenciesBlockParserPlugin {
+				Object.create(this, {
+					parser: { value: parser }
+				})
+			)
+		);
+	}
+
+	processArray(expr, param) {
+		const parser = this.parser;
+		if (param.isArray()) {
+			for (const p of param.items) {
+				const result = this.processItem(expr, p);
+				if (result === undefined) {
+					this.processContext(expr, p);
+				}
+			}
+			return true;
+		} else if (param.isConstArray()) {
+			const deps = [];
+			for (const request of param.array) {
+				let dep, localModule;
+				if (request === "require") {
+					dep = "__webpack_require__";
+				} else if (["exports", "module"].includes(request)) {
+					dep = request;
+				} else if (
+					(localModule = LocalModulesHelpers.getLocalModule(
+						parser.state,
+						request
+					))
+				) {
+					// eslint-disable-line no-cond-assign
+					dep = new LocalModuleDependency(localModule);
+					dep.loc = expr.loc;
+					parser.state.current.addDependency(dep);
+				} else {
+					dep = this.newRequireItemDependency(request);
+					dep.loc = expr.loc;
+					dep.optional = !!parser.scope.inTry;
+					parser.state.current.addDependency(dep);
+				}
+				deps.push(dep);
+			}
+			const dep = this.newRequireArrayDependency(deps, param.range);
+			dep.loc = expr.loc;
+			dep.optional = !!parser.scope.inTry;
+			parser.state.current.addDependency(dep);
+			return true;
+		}
+	}
+	processItem(expr, param) {
+		const parser = this.parser;
+		if (param.isConditional()) {
+			for (const p of param.options) {
+				const result = this.processItem(expr, p);
+				if (result === undefined) {
+					this.processContext(expr, p);
+				}
+			}
+			return true;
+		} else if (param.isString()) {
+			let dep, localModule;
+			if (param.string === "require") {
+				dep = new ConstDependency("__webpack_require__", param.string);
+			} else if (param.string === "module") {
+				dep = new ConstDependency(
+					parser.state.module.buildInfo.moduleArgument,
+					param.range
+				);
+			} else if (param.string === "exports") {
+				dep = new ConstDependency(
+					parser.state.module.buildInfo.exportsArgument,
+					param.range
+				);
+			} else if (
+				(localModule = LocalModulesHelpers.getLocalModule(
+					parser.state,
+					param.string
+				))
+			) {
+				// eslint-disable-line no-cond-assign
+				dep = new LocalModuleDependency(localModule, param.range);
+			} else {
+				dep = this.newRequireItemDependency(param.string, param.range);
+			}
+			dep.loc = expr.loc;
+			dep.optional = !!parser.scope.inTry;
+			parser.state.current.addDependency(dep);
+			return true;
+		}
+	}
+	processContext(expr, param) {
+		const dep = ContextDependencyHelpers.create(
+			AMDRequireContextDependency,
+			param.range,
+			param,
+			expr,
+			this.options
+		);
+		if (!dep) return;
+		dep.loc = expr.loc;
+		dep.optional = !!this.parser.scope.inTry;
+		this.parser.state.current.addDependency(dep);
+		return true;
+	}
 
+	processCallRequire(expr) {
 		const processArrayForRequestString = param => {
```

Comment:
For consistency `processArrayForRequestString` and `processItemForRequestString` should also be member methods.